site/src/Controller/UserController.php

```diff
@@ -223,4 +228,146 @@ public function resendConfirmationEmailAction(
+    {
+        $this->denyAccessUnlessGranted('IS_AUTHENTICATED_FULLY');
+        $user = $this->getUser();
+        $notifications = $notificationManager->getNotifications($user);
+
+        return $this->render(
+            'user/notification_inbox.html.twig',
+            [
+                'notificationList' => $notifications,
+            ]
+        );
+    }
+
+    /**
+     * Mark a notification as seen in the notification inbox
+     *
+     * @Route("/user/notification/seen/{notification}", name="inbox_mark_seen")
+     *
+     * @param Notification        $notification
+     * @param NotificationManager $notificationManager
+     *
+     * @throws \Doctrine\ORM\NonUniqueResultException
+     * @throws \Doctrine\ORM\EntityNotFoundException
+     * @throws \Exception
+     *
+     * @return Response
+     */
+    public function inboxMarkSeenAction($notification, NotificationManager $notificationManager)
+    {
+        $this->denyAccessUnlessGranted('IS_AUTHENTICATED_FULLY');
+
+        $user = $this->getUser();
+
+        $notifiableEntity = $notificationManager->getNotifiableEntity($user);
+        $notifiable = $notificationManager->getNotifiableInterface($notificationManager->getNotifiableEntityById($notifiableEntity));
+
+        $notificationManager->markAsSeen(
+            $notifiable,
+            $notificationManager->getNotification($notification),
+            true
+        );
+        $notificationList = $notificationManager->getNotifications(
+            $notifiable
+        );
+
+        return $this->redirectToRoute(
+            'notification_inbox',
+            ['notificationList' => $notificationList]
+        );
+    }
+
+    /**
+     * Remove a Notification
+     *
+     * @Route("/user/notification/remove/{notification}", name="remove_notification")
+     *
+     * @param Notification        $notification
+     * @param NotificationManager $notificationManager
+     *
+     * @throws \Doctrine\ORM\OptimisticLockException
+     *
+     * @return Response
+     */
+    public function removeAction($notification, NotificationManager $notificationManager)
+    {
+        $this->denyAccessUnlessGranted('IS_AUTHENTICATED_FULLY');
+
+        $user = $this->getUser();
+
+        $notifiableUser = $notificationManager->getNotifiableEntity($user);
+        $notifiableEntity = $notificationManager->getNotifiableEntityById($notifiableUser);
+        $notifiable = $notificationManager->getNotifiableInterface($notifiableEntity);
+
+
+        // Deletes the link between notification and notifiable
+        $notificationManager->removeNotification(
+            [$notifiable],
+            $notificationManager->getNotification($notification),
+            true
+        );
+
+        // Remove the Notification from the database
+        // TODO: Let foreign constraints take care of removing notifications, see #400
+        // $notificationManager->deleteNotification($notificationManager->getNotification($notification), true);
```

Comment:
Wait: you commented this out, so it works without this line?

Reply:
Yup, it does. As I said yesterday removeNotification() removes link between notifiableEntity and notifiableNotification, so if that is removed the user can't see it anymore

Reply:
@imphil The deleteNotification() method is for removing the notification from the database itself

Reply:
It is a workaround probably

Reply:
So the notification is still half-present in the database? That's not what we want, the notification should be fully gone. Sothe deleteNotification() part is necessary.

Reply:
@imphil Exactly!